.github/workflows/docs.yaml

```
@@ -0,0 +1,67 @@
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "!**-rc"
+    paths:
+      - docs/**
+      - .github/workflows/docs.yml
+
+
+name: Deploy Datafusion Python site
+
+jobs:
+  build-docs:
+    name: Build docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set target branch
+        id: target-branch
+        run: |
+          if test '${{ github.ref }}' = 'refs/heads/master'; then
+            echo "value=asf-staging" >> $GITHUB_OUTPUT
+          elif test '${{ github.ref_type }}' = 'tag'; then
+            echo "value=asf-site" >> $GITHUB_OUTPUT
+          fi
+      - name: Checkout docs sources
+        uses: actions/checkout@v3
+      - name: Checkout docs target branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ steps.target-branch.outputs.value }}
+          path: docs-target
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install dependencies
+        run: |
+          python3 -m venv venv
+          source venv/bin/activate
+          pip install -r requirements-310.txt
+          pip install -r docs/requirements.txt
+      
+      - name: Build Datafusion
+        run: |
+          source venv/bin/activate
+          maturin develop
+
+      - name: Build docs
+        run: |
+          source venv/bin/activate
+          cd docs
+          make html
+
+      - name: Copy & push the generated HTML
+        run: |
+          cd docs-target
+          find ./ -name "*.html" -type f -print0 | xargs -0 /bin/rm -f
+          cp -r ../docs/build/html/* .
+          git config --global user.name 'GitHub Actions'
+          git config --global user.email 'actions@github.com'
```

```suggestion
          git config user.name "github-actions[bot]"
          git config user.email "github-actions[bot]@users.noreply.github.com"
```

We can use https://avatars.githubusercontent.com/in/15368?s=40&v=4 icon with the email. See also: https://github.com/ruby/csv/commits/gh-pages

---

I am not sure how to do this.
I just followed the way the cookbook workflow does it.

---

This is how ruby/csv does it: https://github.com/ruby/csv/blob/master/.github/workflows/test.yml#L126-L127
I will changed it to these values too!